### DISCUSSION TITLE
You might not need... bootcamp!

### DISCUSSION THREAD

**Discussion by jamieshark** [community/community#61190] [Programming Help]
**My Story**
I grew up with early exposure to technology and the advent of the Internet, but I never envisioned a career for myself in tech. I consider myself a “nontraditional” developer in the sense that I didn’t take any “formal” computer science classes[^1], but in reality, I had been building relevant skills throughout my whole life. I spent most of college studying quantum physics and molecules while memorizing sushi rolls and customers’ names for my part-time restaurant job. High school was devoted to practicing piano sonatas and violin concertos. Though seemingly unrelated, these were all the foundations of understanding levels of abstraction, cultivating customer relations, working with others under pressure, and recognizing patterns in large systems. Maybe my training started even further back with childhood, wherein I spent hours constructing Lego submarine research stations or entering precarious hex codes into my GameShark? I’ll let you decide.

I was laid off from my first job after college, but a few temp jobs and barista shifts later, I found a job as a project manager for a front-end project that was severely understaffed. At some point, I realized that in the time it took for me to write the issue, assign it to someone, and wait for the fix, I could dig in my heels and fix it myself[^2]. Copy changes turned into basic code functions, issues became commit messages, and within a few months, I was spending less time frantically searching [Codecademy](https://www.codecademy.com/) and more time shipping to production, becoming curious about code patterns, and even contributing to a handful of open-source projects (if you’re on GitHub, chances are you know of a few that [could use your help](https://github.com/topics/good-first-issue)!). Eventually, the tech lead asked to have me as a full-time developer on the project. Years later, I’m pressed to imagine a better career fit for my personality and interests.

**No Ragrets [sic]**
When people ask me whether I regret not having gone to bootcamp, I recognize the “Need Job Experience to Get The Job to Get Experience” loop is difficult to overcome, and bootcamps offer a path around that trap, but no amount of school will teach you the intangible skills you might need as a developer[^3]. While I would have appreciated a more formal learning structure, following the momentum of learning on the job felt easier than enrolling in a bootcamp (which were not as popular/available at the time).

What I’m trying to say is, “[It depends.](https://www.youtube.com/watch?v=aMWh2uLO9OM)” I think the most important thing someone can take away from any kind of schooling is uncovering how they learn, in addition to what, whether that is in a classroom with others, through late-night solo reading sessions, or something else that works better for them. Just based on the number of apps and productivity tools out there, you can guess that one thing you’re constantly doing in tech is learning new stuff. Every team and every company might use the same programming language in a different way, so understanding the core building blocks and principles of programming is going to serve you better in the long run than chasing after learning the latest technology[^4]. 

**All Roads Lead to `/usr/home/`**
While many people have the impression that programming is strictly a STEM-adjacent field, I have found that great code resembles poetry more than anything else, which is _why I vehemently stress that there are multiple paths to being a good engineer and no one should count themselves out_.That is also not to say the only way to work in tech is to be a software engineer and vice versa; there are a variety of industries in need of engineers and even more open-source projects that need design or management contributions. Get your foot in the door by focusing on transferable skills and demonstrating work. What gets me excited about someone is their story from previous roles and responsibilities, especially if it comes from a non-technical background or industry. GitHub even has a way to [show your story](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/customizing-your-profile/managing-your-profile-readme) alongside your work. The critical point is to shift the narrative to highlight your new goals and make yourself irreplaceable because of your unique background. 

Once you’ve got your foot in the door, wedge it open by finding people who will vouch for you and your ambitions. Find organizations and opportunities that firmly believe “nontraditional” does not equal “unqualified.” Lastly, keep the door open for those coming in behind you. Combat imposter syndrome by recognizing actual work being done (regardless of someone’s title) and giving credit where it is due: there is no easy work in this world and we all deserve to do work we enjoy with people we love, and it is Very Brave to go after that for ourselves.

[^1]: Even though the term “nontraditional” is used to signify some lack of background in computer science / web development classes, I still believe the opportunity to go to school (at all) and be adjacent to these fields is a huge advantage.
[^2]: In hindsight, this might have been a bit of a reach since my most relevant experience was having a Xanga in 2000.
[^3]: The terms “developer” and “engineer” are used pretty interchangeably in this post. I am still not totally sure what to call myself just yet, but I usually go with what sounds fanciest.
[^4]: The title of this discussion is a reference to https://youmightnotneed.com/, a site that does a great job of demonstrating this.

**Comment by kaladivyaprada** [comment]
Adda

**Comment by kaladivyaprada** [comment]
Good

**Comment by joan-plazas** [comment]
👍

**Comment by ghostinhershell** [comment]
Love this, thank you for sharing @jamieshark ❤️

**Comment by rock-23-s** [comment]
👍

**Comment by kjburton03** [comment]
I wish I could say I was self taught, but I did not have any structure until going to a boot camp in Nashville.  Plus it provided me an amazing support system. It truly changed my life for the better.

But with that being said,  you are correct, bootcamps don't teach you half of what you need to know and it's mostly hands off teaching unless you encounter a problem. Honestly the main thing keeping me going through bootcamp until i got into a routine was how much money I spent/how many people knew I was in school.

**Reply by jamieshark** [reply]
Yes, I think the main idea is to do what works best for you with the resources you have available! I'm glad you found something that worked for you and got you closer to where you wanted to be.

**Comment by chat14** [comment]
Bulcredit uk
           https://creditblu.com/#a_aid=96997&a_bid=b9bfa533&data1=RentaCompany

**Comment by yfming93** [comment]
Good good study, day day up.

**Comment by hugo-sid** [comment]
> What gets me excited about someone is their story from previous roles and responsibilities

Thanks for sharing your unique journey @jamieshark .

**Comment by 1Haschwalth** [comment]
good job！

**Comment by Smi41** [comment]
Please help me.

**Reply by Smi41** [reply]
Hi

**Comment by zeddd25** [comment]
Nice!

**Comment by kamruzzaman-dev** [comment]
Thank you for sharing!
